fl4health/strategies/feddg_ga.py

```diff
@@ -220,18 +231,41 @@ def configure_fit(
 
         self.initial_adjustment_weight = 1.0 / len(client_fit_ins)
 
-        # Setting self.num_rounds
+        # Setting self.num_rounds once and doing some sanity checks
+        assert self.on_fit_config_fn is not None, "on_fit_config_fn must be specified"
+        config = self.on_fit_config_fn(server_round)
+        assert "evaluate_after_fit" in config, "evaluate_after_fit must be present in config"
+        assert config["evaluate_after_fit"] is True, "evaluate_after_fit must be set to True"
+
+        assert "pack_losses_with_val_metrics" in config, "pack_losses_with_val_metrics must be present in config"
+        assert config["pack_losses_with_val_metrics"] is True, "pack_losses_with_val_metrics must be set to True"
+
+        assert "n_server_rounds" in config, "n_server_rounds must be specified"
+        assert isinstance(config["n_server_rounds"], int), "n_server_rounds is not an integer"
+
         if self.num_rounds is None:
-            assert self.on_fit_config_fn is not None, "on_fit_config_fn must be specified"
-            config = self.on_fit_config_fn(server_round)
-            assert "evaluate_after_fit" in config, "evaluate_after_fit must be present in config and set to True"
-            assert config["evaluate_after_fit"] is True, "evaluate_after_fit must be set to True"
-            assert "n_server_rounds" in config, "n_server_rounds must be specified"
-            assert isinstance(config["n_server_rounds"], int), "n_server_rounds is not an integer"
             self.num_rounds = config["n_server_rounds"]
+        else:
+            assert config["n_server_rounds"] == self.num_rounds, "n_server_rounds has changed from the original value"
```

Comment:
minor: might be nice to include the values of `config["n_server_rounds"]` and `n_server_rounds` in the assertion

Reply:
Yup, makes sense!